File: javaToC++/DFunction.cpp

/*
Class: CPSC 122-01
Team Member 1: Paul De Palma
Team Member 2: None
Submitted By: Paul De Palma
GU Username: depalma
File Name: DFunction.cpp
Program illustrates
  void function 
  while loop
  modulus
To Build: g++ DFunction.cpp -o DFunction    //DFunction names executable 
To Execute: ./DFunction
*/


#include <iostream>
using namespace std;

void display(int, int);

int main()
{
 int num, col; 

 cout << "How many integers do you want to see?" << endl; 
 cin >> num;

 cout << "Distributed over how many columns" << endl; 
 cin >> col;

 display(num,col);
 cout << endl;

 return 0;
} 

void display(int nums, int cols)
{
 int ct = 0;
 int rowCt = 1;
 
 while (ct < nums)
  {
   cout << ct << '\t';         //tab to next col 
   if (ct % cols == cols - 1)  //think about this one 
    cout << endl;
   ct++;
  }
}

File: javaToC++/DFunction.java
/*
Class: CPSC 122
Section: 01
Team Member 1: Paul De Palma
Team Member 2: None
Submitted By: Paul De Palma
GU Username: depalma
File Name: DFunction.java
Program illustrates 
void function
while loop
modulus
To Build: javac DFunction.java
To Execute: java DFunction 
*/

import java.util.Scanner;

public class DFunction 
{
  public static void main(String[] args)
  {
    int num, cols; 

    Scanner keyboard = new Scanner(System.in);


    System.out.println("How many integers do you want to see?");
    num = keyboard.nextInt();

    System.out.println("Distributed over how many columns");
    cols = keyboard.nextInt();

    display(num,cols);

  }

 public static void display(int num, int cols)
  {
    int ct = 0;
    int rowCt =1;

    while (ct < num)
     {
       System.out.print(ct);
       System.out.print('\t');
       if (ct % cols == cols - 1)
	 System.out.println();
       ct++;
     }
    System.out.println();
  }
} 
	
